packages/react-router/lib/dom/ssr/single-fetch.tsx

```diff
@@ -694,12 +708,14 @@ function unwrapSingleFetchResult(
   }
 
   let routeResult = result.routes[routeId];
-  if ("error" in routeResult) {
+  if (routeResult == null) {
+    throw new Error(`No result found for routeId "${routeId}"`);
```

Comment:
Provide a better error message if this happens than the vague "cannot look for error in undefined" message users get today